apps/whale-api/src/module.api/consortium.service.ts

```diff
@@ -1,17 +1,129 @@
+    const attrs = (await this.rpcClient.masternode.getGov('ATTRIBUTES')).ATTRIBUTES
+    const searching: boolean = searchTerm !== ''
+    const txIdFormatRegex: RegExp = /^[a-z0-9]{64}$/
+    const membersKeyRegex: RegExp = /^v0\/consortium\/\d+\/members$/
+    const searchForTxId = searching && txIdFormatRegex.exec(searchTerm) !== null
+    const searchForMemberDetail = searching && txIdFormatRegex.exec(searchTerm) === null
+    let totalTxCount: number = 0
+
+    const members = (Object.entries(attrs) as [[string, object]]).reduce((prev: MemberDetail[], [key, value]) => {
+      if (membersKeyRegex.exec(key) === null) {
+        return prev
+      }
+
+      (Object.entries(value) as [[string, MemberDetail]]).forEach(([memberId, memberDetail]) => {
+        if (searchForMemberDetail) {
+          if (!(memberDetail.ownerAddress === searchTerm || memberDetail.name.toLowerCase().includes(searchTerm))) {
+            return prev
+          }
+        }
+
+        if (!prev.some(m => m.id === memberId)) {
+          prev.push({
+            id: memberId,
+            name: memberDetail.name,
+            ownerAddress: memberDetail.ownerAddress
+          })
+        }
+      })
+
+      return prev
+    }, [])
+
+    if (searchForTxId) {
+      const foundTx = await this.transactionMapper.get(searchTerm)
+      if (foundTx !== undefined) {
+        const relevantTxsOnBlock = await this.rpcClient.account.listAccountHistory(members.map(m => m.ownerAddress), {
+          maxBlockHeight: foundTx.block.height,
+          depth: 0,
+          txtypes: [DfTxType.MINT_TOKEN, DfTxType.BURN_TOKEN]
+        })
+        const transaction = relevantTxsOnBlock.find(tx => tx.txid === foundTx.txid)
+
+        if (transaction === undefined) {
+          return {
+            total: 0,
+            transactions: []
+          }
+        }
+
+        return {
+          total: 1,
+          transactions: [this.formatTransactionResponse(transaction, members)]
+        }
+      }
+
+      return {
+        total: 0,
+        transactions: []
+      }
+    }
+
+    const transactions: AccountHistory[] = await this.rpcClient.account.listAccountHistory(members.map(m => m.ownerAddress), {
+      txtypes: [DfTxType.MINT_TOKEN, DfTxType.BURN_TOKEN],
+      including_start: true,
+      start: pageIndex * limit,
+      limit
+    })
+
+    const promises = []
+    for (let i = 0; i < members.length; i++) {
+      promises.push(this.rpcClient.account.historyCount(members[i].ownerAddress, { txtype: DfTxType.BURN_TOKEN }))
+      promises.push(this.rpcClient.account.historyCount(members[i].ownerAddress, { txtype: DfTxType.MINT_TOKEN }))
+    }
+    const counts = await Promise.all(promises)
+    totalTxCount = counts.reduce((prev, curr) => {
+      return prev + curr
+    }, 0)
+
+    return {
+      transactions: transactions.map(tx => {
+        return this.formatTransactionResponse(tx, members)
+      }),
+      total: totalTxCount
+    }
+  }
```

Comment:
suggest breaking this down into several smaller functions to improve readability as this function is rather long and complex

Reply:
Done here: [`8789df4`](https://github.com/JellyfishSDK/jellyfish/pull/1868/commits/8789df44e480fc65cb65919d6f8eefa00bbe6fe5)